pkg/operator/ceph/operator.go

```diff
@@ -129,12 +132,15 @@ func (o *Operator) Run() error {
 	var namespaceToWatch string
 	if os.Getenv("ROOK_CURRENT_NAMESPACE_ONLY") == "true" {
 		logger.Infof("Watching the current namespace for a cluster CRD")
-		namespaceToWatch = namespace
+		namespaceToWatch = o.operatorNamespace
 	} else {
 		logger.Infof("Watching all namespaces for cluster CRDs")
 		namespaceToWatch = v1.NamespaceAll
 	}
 
+	// Start the controller-runtime Manager.
+	go o.startManager(stopChan)
```

Comment:
The final change I'd like to see in this PR (assuming the follow-up PRs mentioned), is that the canary pods not be started unless a cluster is created that has `managePodBudgets: true`. This might be achieved by moving this line down into the `startSystemDaemons()` callback and modifying that callback to take a parameter indicating the new setting.